monitoring/reporter/expvar/expvars.go

```diff
@@ -0,0 +1,92 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package expvar
+
+import (
+	"errors"
+	"expvar"
+	"fmt"
+	"net/http"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+	"github.com/elastic/elastic-agent-libs/transform/typeconv"
+	"github.com/elastic/elastic-agent-shipper/monitoring/reporter"
+)
+
+//Config is the config struct for marshalling whatever we get from the config file
+type Config struct {
+	Enabled bool   `config:"enabled"`
+	Host    string `config:"Host"`
+	Port    int    `config:"port"`
+	Name    string `config:"name"`
+}
+
+// Expvars is the simple manager for the expvars web interface
+type Expvars struct {
+	log     *logp.Logger
+	metrics reporter.QueueMetrics
+	srv     *http.Server
+}
+
+// NewExpvarReporter initializes the expvar interface, and starts the http frontend.
```

Comment:
[Nit | suggestion]
from my perspective a concrete type, that implements an interface is initializes, not the interface itself as the test here suggests.
```suggestion
// NewExpvarReporter initializes the expvar reporter, and starts the http frontend.
```